src/earn/EarnPoolInfoScreen.tsx

```diff
@@ -491,6 +499,72 @@ export default function EarnPoolInfoScreen({ route, navigation }: Props) {
       .filter((token): token is TokenBalance => !!token)
   }, [tokens, allTokens])
 
+  const depositToken = allTokens[dataProps.depositTokenId]
+  if (!depositToken) {
+    // This should never happen
+    throw new Error(`Token ${dataProps.depositTokenId} not found`)
+  }
+
+  const { swappableFromTokens } = useSwappableTokens()
+  const cashInTokens = useCashInTokens()
+  const isSwapEnabled = useSelector(isAppSwapsEnabledSelector)
+  const userLocation = useSelector(userLocationDataSelector)
+
+  const canDeposit = useMemo(() => {
+    return allTokens[dataProps.depositTokenId]?.balance?.gt(0) ?? false
+  }, [pool, allTokens])
+  const canSameChainSwapToDeposit = useMemo(() => {
+    return (
+      isSwapEnabled &&
+      !!swappableFromTokens.find(
+        (tokenInfo) =>
+          tokenInfo.networkId === networkId && tokenInfo.tokenId !== dataProps.depositTokenId
+      )
+    )
+  }, [pool, isSwapEnabled, swappableFromTokens])
+  const canCrossChainSwapToDeposit = useMemo(() => {
+    return (
+      getFeatureGate(StatsigFeatureGates.ALLOW_CROSS_CHAIN_SWAPS) &&
+      isSwapEnabled &&
+      !!swappableFromTokens.find((tokenInfo) => tokenInfo.networkId !== networkId)
+    )
+  }, [pool, isSwapEnabled, swappableFromTokens])
+  const canCashIn = useMemo(() => {
+    return !!cashInTokens.find((tokenInfo) => tokenInfo.tokenId === dataProps.depositTokenId)
```

Comment:
Can you put all of this into its own hook like useDepositEntrypointInfo or something that returns all of the props for BeforeDepositBottomSheet?